src/Core/Batch/BatchRunner.php

```diff
@@ -0,0 +1,183 @@
+     *     @type int $workerNum The number of child processes. It only takes
+     *               effect with the {@see \Google\Cloud\Core\BatchDaemon}.
+     *     @type string $bootstrapFile A file to load before executing the
+     *                  job. It's needed for registering global functions.
+     * }
+     * @return boolean true on success, false on failure
+     * @throws \InvalidArgumentException When receiving a Closure.
+     */
+    public function registerJob($identifier, $func, $options = [])
+    {
+        if ($func instanceof \Closure) {
+            throw new \InvalidArgumentException('Closure is not allowed');
+        }
+        // Always work on the latest data
+        $result = $this->configStorage->lock();
+        if ($result === false) {
+            return false;
+        }
+        $this->config = $this->configStorage->load();
+        $this->config->registerJob($identifier, $func, $options);
+
+        $result = $this->configStorage->save($this->config);
+        if ($result === false) {
+            return false;
+        }
+        $this->configStorage->unlock();
+        return true;
+    }
+
+    /**
+     * Submit an item.
+     *
+     * @param string $identifier Unique identifier of the job.
+     * @param mixed $item It needs to be serializable.
+     *
+     * @return boolean true on success, false on failure
+     */
+    public function submitItem($identifier, $item)
+    {
+        $job = $this->getJobFromId($identifier);
+        if ($job === null) {
+            throw new \RuntimeException(
+                'The identifier does not exist: ' . $identifier
+            );
+        }
+        $idNum = $job->getIdnum();
+        return $this->submitter->submit($item, $idNum);
+    }
+
+    /**
+     * Get the job with the given identifier.
+     *
+     * @param string $identifier Unique identifier of the job.
+     *
+     * @return BatchJob|null
+     */
+    public function getJobFromId($identifier)
+    {
+        return $this->config->getJobFromId($identifier);
+    }
+
+    /**
+     * Get the job with the given numeric id.
+     *
+     * @param int $idNum A numeric id of the job.
+     *
+     * @return BatchJob|null
+     */
+    public function getJobFromIdNum($idNum)
+    {
+        return $this->config->getJobFromIdNum($idNum);
+    }
+
+    /**
+     * Get all the jobs.
+     *
+     * @return BatchJob[]
+     */
+    public function getJobs()
+    {
+        return $this->config->getJobs();
+    }
+
+    /**
+     * Load the config from the storage.
+     *
+     * @return boolean true on success, false on failure
+     */
+    public function loadConfig()
+    {
+        $result = $this->configStorage->lock();
+        if ($result === false) {
+            return false;
+        }
+        $result = $this->configStorage->load();
+        $this->configStorage->unlock();
+        if ($result === false) {
+            // TODO: think what to do
```

Comment:
Just a friendly reminder about this guy - if it is something we would like to follow up on later, can we create an issue to track the work?